docs/pages/reference/helm-reference/teleport-kube-agent.mdx

```diff
@@ -804,6 +925,31 @@ Enables the creation of a Kubernetes persistent volume to hold Teleport instance
 
 [Kubernetes reference](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)
 
+#### Chart upgrades
+
+If `storage.enabled` is `false` and you are upgrading a legacy
+`teleport-kube-agent` chart, the chart will launch a Teleport pod as a
+`Deployment` instead of a `StatefulSet`, disabling persistent storage for the
+Teleport pod. If `storage.enabled` is set to `true` during an upgrade, the chart
+will create a webhook that deletes any legacy `Deployment` and launches a
+`StatefulSet` instead.
```

Comment:
Depending on the storage value, the upgrade process is different:

# `storage.enabled=false`

Previously, the Teleport pod was part of a `Deployment` resource.
During the upgrade, a `Statefulset` resource is installed alongside the `Deployment`.
Once the Statefulset joins the cluster and becomes healthy, the chart deletes the `Deployment` object using a Helm Hook, and the upgrade finishes.

# `storage.enabled=true`
Previously, the Teleport pod was part of a `StatefulSet` resource.
During the upgrade, the resource is updated (kept as `StatefulSet`), and the identity is imported from external storage into the Kubernetes secret.